### PR TITLE
simplify meta-data usage in cpp

### PR DESCRIPTION
**Pull request by Disservin**
- directly parse the json information into the struct

Code will fail, when the json is missing a key, can that be the case?

### CODE REVIEW

**Comment by robertnurnberg** [comment]
Yes. See usage of `--onlySPRT` switch.

**Comment by Disservin** [comment]
Mh true, I'll try to make it work with optional keys.

**Comment by vondele** [comment]
this one will need a rebase

**Comment by robertnurnberg** [comment]
Did you check for non-functionality? ;)

I get a nearly emtpy `updateWDL.json` file for the update script, locally.

File: scoreWDLstat.cpp [hunk 1]
```diff
@@ -175,16 +174,17 @@ void ana_files(map_t &map, const std::vector<std::string> &files, const std::str
                 std::exit(1);
             }
 
-            if (meta_map.at(test_filename).find("book_depth") != meta_map.at(test_filename).end()) {
-                std::string book_depth = meta_map.at(test_filename)["book_depth"];
-                move_counter           = std::to_string(std::stoi(book_depth) + 1);
+            if (meta_map.at(test_filename).book_depth.has_value()) {
+                auto book_depth = meta_map.at(test_filename).book_depth.value();
```
Comment (robertnurnberg) [inline scoreWDLstat.cpp:178]: I did this over two lines in master, to avoid having to cast to string in the second line. Can now be put into line I think.

**Comment by Disservin** [comment]
> Did you check for non-functionality? ;)
> 
> I get a nearly emtpy `updateWDL.json` file for the update script, locally.

Mh let me check..

**Pull request merged**